### DISCUSSION TITLE
Dimensionality issues in transfrom

### DISCUSSION THREAD

**Discussion by shenao1995** [Project-MONAI/tutorials#918] [General] [answered]
My monai version is 0.8.1. I am working on a classification task for 3d images. I am making predictions by resizing the image to (96,96,32). Why does it always report the error that the size is not the same when I run it. The specific error is: **RuntimeError: stack expects each tensor to be equal size, but got [1, 96, 96, 32] at entry 0 and [1, 32, 96, 96] at entry 1.** Where I'm confused is why is 32 sometimes on x and sometimes on z? My transform code is as follows: 
```
train_transforms = Compose(
        [
            LoadImaged(keys=modalKey[:-1]),
            EnsureChannelFirstd(keys=modalKey[:-1]),
            Resized(keys=modalKey[:-1], spatial_size=(input_size, input_size, 32), mode="trilinear",
                    align_corners=True),
            ConcatItemsd(keys=modalKey[:-1], name="inputs"),
            ScaleIntensityd(keys=["inputs"]),
            RandGaussianNoised(keys=["inputs"], prob=0.5),
            RandRotate90d(keys=["inputs"], prob=0.5, spatial_axes=(0, 2)),
            RandAffined(keys=["inputs"], prob=0.5, scale_range=[0.1, 0.2],
                        translate_range=(30, 40), rotate_range=(np.pi / 5, np.pi / 5),
                        padding_mode="border"),
            RandZoomd(keys=["inputs"], prob=0.5, max_zoom=2, min_zoom=0.1),
            ToTensord(keys=["inputs"]),
        ]
    )
```
I would like to ask the developers or other experienced partners to help me out, thanks!

**Comment by KumoLiu** [comment, marked as answer]
Hi @shenao1994, it seems you had set the `spatial_axes` in `RandRotate90d ` incorrectly. The `spatial_axes` defines the plane to rotate with 2 spatial axes. (0, 2) means rotate in the XZ plane which will invert the dimensions of x and z, so you can simplify set it as (0,1). 
Hope it can help you! If you have any other problem, free feel to ask me!

**Reply by shenao1995** [reply]
The program no longer reports errors. Thank you!
